### DISCUSSION TITLE
Custom function to add consecutive dates

### DISCUSSION THREAD

**Discussion by Mirodin** [orbitalquark/textadept#349] [Q&A] [answered]
I want to leverage a custom function to use a text file as a [calendar.txt](https://terokarvinen.com/2021/calendar-txt/). Therefore when having the file open in Textadept I would like to open the command prompt (`ctrl+E`) and run i.e. `addWeek`.

```
2023-01-08 w01 Sun
2023-01-07 w01 Sat
2023-01-06 w01 Fri
...
```

Would become this:

```
2023-01-15 w02 Sun
2023-01-14 w02 Sat
2023-01-13 w02 Fri
2023-01-12 w02 Thu
2023-01-11 w02 Wed
2023-01-10 w02 Tue
2023-01-09 w02 Mon
2023-01-08 w01 Sun
2023-01-07 w01 Sat
2023-01-06 w01 Fri
...
```

This is what I have so far I am just missing the "find the last date" part.

```lua
function addWeek()
	local lastDate = -- get last date i.e. 2023-01-01
	local year, month, day = string.match(lastDate, "(%d+)-(%d+)-(%d+)")
	local date = { year = year, month = month, day = day }
	local tmp = {}
	for i = 1, 7 do
		date.day = date.day + 1
		local ts = os.time(date)
		date = os.date("*t", ts)
		tmp[#tmp + 1] = os.date("%Y-%m-%d w%W %a", ts)
	end
	buffer.document_start()
	buffer.add_text(table.concat(tmp, "\n") .. "\n")
end
```

Thanks in advance :)

**Comment by orbitalquark** [comment, marked as answer]
If your most recent week is at the top, then maybe something like this:

     local year, month, day, week = 
buffer:get_line(1):match('(%d+)%-(%d+)%-(%d+)')
     if not year then return end
     local lastDate = os.time{year = year, month = month, day = day}
     local date = os.date('*t', lastDate)
     ...

**Reply by Mirodin** [reply]
Thank you, works like a charm :)
